src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSourceRequest.java

```diff
@@ -115,7 +116,10 @@
     @CheckForNull
     private Iterable<BitbucketBranch> branches;
     // TODO private Iterable<BitbucketTag> tags;
-
+    /**
+     * The BitbucketApi that is used for the request.
+     */
+    private BitbucketApi api;
```

Comment:
I think this is not needed.

AFAICT this is intended to be [used by the extension to recover the PR info](https://github.com/kylec32/bitbucket-scm-filter-approval/blob/master/src/main/java/tk/lightweightcoding/jenkins/bitbucket/approval/ApprovedPullRequestTrait.java#L100). However required info can be just added to the original `BitbucketPullRequest` (as suggested by @casz) and then used in the extension. This way no new API call is needed.

Also, I'd like to keep `BitbucketApi` as _private_ as possible. I know it is technically public but as currently designed, extensions do not have the need to use it (and it should remain like that).

Reply:
I do see the desire to keep this as private as possible but as described in the comment above the `BitbucketPullRequest` that is passed to me doesn't have the information I need (reviewers) and I don't have access to the credentials or anything like that to make the request myself so this is what I ended up with.

Definitely open for suggestions.